src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than zero');
+    }
+
+    const sides = [a, b, c].sort((x, y) => x - y);
+
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
+    }
+
+    this.color = color;
```

Comment:
[CODE KNOWLEDGE] - Since 'color' is already declared with an access modifier in the constructor, you don't need to manually assign it to 'this.color'. TypeScript does this automatically.